README
/*
	README - 31.01.2011

	phpRechnung - is easy-to-use Web-based multilingual accounting software.
	Copyright (C) 2001 - 2011 Edy Corak < edy at loenshotel dot de >

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation; either version 2 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/


	Um phpRechnung 1.6.4 zu installieren wird folgendes benoetigt:

	------
	Server
	------

	Webserver:
	Apache, Lighttpd oder auch ( Microsoft IIS - nicht getestet )

	PHP:
	PHP Version 5 - ( mbstring, PEAR::Mail, Mail_Mime )

	MySQL: ( ab phpRechnung 1.8 auch andere Datenbaksysteme )
	MySQL Version 5 oder 4

	MailServer:
	Postfix, Exim, Sendmail...

	Die oben genannten Programme muessen bereits
	installiert und konfiguriert sein.

	------
	Klient
	------

	Browser:
	z. B. Firefox, Konqueror, Safari, Opera, Google Chrome ...
	( Javascript muss eingeschaltet sein )

	PDF-Betrachter

	------

	Wichtig:
	Unter Einstellung muss eine korrekte E-Mail Adresse eingetragen
	werden, da ansonsten der E-Mail Verasand nicht funktioniert.
	Das selbe gilt auch, wenn ein Relay-Server benutzt wird,
	da ansonsten die E-Mail's vom Relay-Server abgewiesen werden.

	Wenn Sie keinen eigenen E-Mail Server haben, dann stellen Sie
	bitte in der Datei /include/phprechnung.inc.php die Variable
	$PHPSendMail = ""; auf $PHPSendMail = "1"; ein.

	In der Datei /include/mail.inc.php muessen noch
	Aenderungen vorgenommen werden Host, Port und falls
	Anmeldung notwendig ist, $Smtp["auth"] auf true zu stellen
	und $Smtp["username"] und $Smtp["password"] eintragen.

	$Smtp["host"] = "ihrmailserver";
	$Smtp["port"] = "25";
	$Smtp["auth"] = false;
	$Smtp["username"] = "";
	$Smtp["password"] = "";

	#####

	Wichtiger Hinweis:

	Bitte haben Sie Verstaendnis, dass fuer die oben genannten
	Programme keine Anfragen mehr beantwortet werden koennen.

	Fragen zu ( open_basedir, safe_mode etc. ) haben nichts mit
	phpRechnung zu tun und werden auch nicht mehr beantwortet
	
	#####

	phpRechnung_1_6_4.tar.gz entpacken - tar -zxvf phpRechnung_1_6_4.tar.gz
	Das Verzeichnis 'phpRechnung_1_6_4' auf den Webserver kopieren
	z. B. /var/www Sie koennen das Verzeichnis auch umbenennen.

	####################

	!!! WICHTIGER HINWEIS !!!

	Seit phpRechnung 1.6 RC1 wird Smarty Template Engine verwendet.
	Damit das ganze funktioniert, muessen noch einige Aenderungen
	vorgenommen werden.
	
	Der Webserver muss fuer die Verzeichnise '/include/smarty/cache'
	und '/include/smarty/templates_c' Schreibrechte bekommen.

	!!! Es wird empfohlen ein Verzeichnis ausserhalb des DocumentRoot zu verwenden !!!

	Das gilt auch fuer die Verzeichnise '/include/smarty/templates' und '/include/smarty/configs'

	Nach den Aenderungen muessen Sie dies auch phpRechnung mitteilen.

	In '/include/smarty.inc.php' folgende Variablen anpassen:
	$Templateroot und $Cacheroot von $_SERVER['DOCUMENT_ROOT']."phpRechnung";
	
	aendern in z. B.

	$smarty->template_dir = "/var/www/include/smarty/templates";
	$smarty->config_dir = "/var/www/include/smarty/configs";
	$smarty->compile_dir = "/var/www/include/smarty/templates_c";
	$smarty->cache_dir = "/var/www/include/smarty/cache";

	Achten Sie bitte auf eine korrekte Schreibweise.

	Mehr Informationen finden Sie unter Link: http://www.smarty.net/

	FAQ Link: http://www.loenshotel.de/phpRechnung/FAQ.php#faq1.4

	####################

	Datenbank erstellen:

	####################

	Wichtiger Hinweis:

	Diese Anleitung ist nur fuer eine Neu-Installation gedacht.

	Wenn Sie ein UPDATE von phpRechnung 1.6 Test3.01 durchfuehren
	moechten, dann lesen Sie bitte in der Datei UPDATE weiter.

	Vor einem UPDATE bitte alle Daten sichern.

	####################

	mysqladmin -u 'benutzername' -p create phprechnung

	Tabellen erstellen:
	mysql -u 'benutzername' -p phprechnung < phprechnung_1_6_4.sql

	Falls kein 'root' Zugriff vorhanden, dann am besten 'phpMyAdmin' benutzen.

	Zuerst Datenbank erstellen, waehlen Sie dann die Datenbank aus, dann Importieren,
	Datei dursuchen, passende SQL Datei auswaehlen: phprechnung_1_6_4.sql,
	Zeichencodierung der Datei 'latin1' waehlen und auf OK klicken.

	Sie sollten dann z. B. folgende Meldung sehen:

	Der Import wurde erfolgreich abgeschlossen, 127 Abfragen wurden ausgefuehrt.

	!!! WICHTIG !!!

	Bitte fuegen Sie NICHT die SQL-Datei in das Feld SQL-Befehl(e) in Datenbank ein.
	Die Daten werden ohne Fehler eingefuegt aber eine Anmeldung ist danach NICHT mehr moeglich.

	Folgende Dateien muessen geaendert werden
	damit man auf die neu erstellte Datenbank 'phprechnung' zugreifen kann.

	Im Verzeichnis 'include'

	'dbconf.php' fuer die Datenbankverbindung
	_DBHOST z. B. "localhost"
	_BBUSER z. B. "benutzername"
	_DBPASS z. B. "benutzerpasswort"
	_DBNAME z. B. "phprechnung"

	Der _DBUSER muss ueber ausreichende Rechte auf dem
	Datenbankserver verfuegen - Erstellen, Aendern, Loeschen etc.

	'phprechnung.inc.php'
	( wichtig ist die Variable "$web" dort bitte eigene Webadresse zu
	phpRechnung eintragen ) z. B. http://ihrserver/phpRechnung
	( ohne '/' am Ende ) da sonst die Weiterleitung nicht funktioniert.

	Fuer die Benutzer die phpRechnung auf einem Windows Server betreiben,
	bitte unter Konfiguration/Einstellung den Wert von TMP-Verzeichnis
	anpassen. Standard ist '/tmp/' in z. B. 'c:\windows\temp\'

	Bitte beachten Sie den letzten Backslash. Wichtig.
	Der Webserver muss Schreibrechte fuer das Verzeichnis haben.
	
	Auf einem Windows Server bitte auch 'session.save_path' ueberpruefen.

	Standard ist session.save_path = /tmp
	Wenn das der Fall ist, dann bitte aendern
	in z. B. session.save_path = c:\windows\temp

	Anmeldung:

	Administrator: admin, admin

	Die Benutzerdaten sind in der Tabelle
	'user' verschluesselt gespeichert

	Nach der Ersten Anmeldung werden Sie aufgefordert die
	phpRechnung Lizenz ( GPL v2 ) zu akzeptieren.
	Dieses Vorgehen ist fuer jeden Benutzer einmalig notwendig.

	Das Administrator-Passwort kann geaendert werden
	nur der Anmeldename 'admin' darf nicht veraendert
	werden. Ansonsten werden viele Seite nicht anzeigen.

	Wenn Sie Fragen oder Anregungen zu phpRechnung haben, dann gehen
	Sie bitte zu Link: http://sourceforge.net/projects/phprechnung/ und
	benutzen das Forum.

	Bitte haben Sie Verstaendnis, dass ich direkte Anfragen zu phpRechnung
	aus Zeitgruenden leider nicht mehr beantworten kann.

	phpRechnung Homepage - http://www.loenshotel.de/phpRechnung/

	Wenn Sie ueber neue Versionen informiert werden moechten, dann
	gehen Sie bitte zu http://freshmeat.net/projects/phprechnung/ und
	lassen sich registieren ( subscribe to new releases )
	oder http://lists.sourceforge.net/lists/listinfo/phprechnung-news

	Vielen Dank fuer die Nutzung von phpRechnung

	Edy Corak < edy at loenshotel dot de >


ChangeLog
/*
	ChangeLog - 31.01.2011

	phpRechnung - is easy-to-use Web-based multilingual accounting software.
	Copyright (C) 2001 - 2011 Edy Corak < edy at loenshotel dot de >

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation; either version 2 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/
	31.01.2011

	Changes in phpRechnung 1.6.4
	---------------------------------------

	- FormatDBNumber has been fixed.

	27.01.2011

	Changes in phpRechnung 1.6.3
	---------------------------------------

	- Problem creating initial cashbook entry was fixed.

	- Simple credit note function was added. You can create invoices with negative total amount.
	  By choosing 'bar/cash' as method of payment, cash book will be also automatically updated.

	- If PrintCompanyData was disabled the Company Data was also not included by sending invoice as PDF-Attachment. fixed.

	- minor bugs were fixed

	05.01.2011

	Changes in phpRechnung 1.6.2
	---------------------------------------

	This is a Bugfix-Only-Release.

	- Last Site Redirection missing query was fixed

	- syslog_syslogid_seq was disabled

	- minor bugs were fixed


	08.12.2010

	Changes in phpRechnung 1.6.1
	---------------------------------------

	This is a Security Bugfix-Only-Release.

	A security issue affects the following phpRechnung releases:

	phpRechnung <= 1.6

	There are security vulnerabilities in phpRechnung 1.6 which
	allow an unauthorized user to have read access of the data.

	It is strongly recommendend to update your phpRechnung <= 1.6 to version 1.6.1 as soon as possible.


	03.12.2010

	Changes in phpRechnung 1.6
	---------------------------------------

	This is a Security Bugfix-Only-Release.

	A security issue affects the following phpRechnung releases:

	phpRechnung <= 1.6 RC2

	It is strongly recommendend to update your phpRechnung <= 1.6 RC2 to version 1.6 as soon as possible.

	There are multiple security vulnerabilities in phpRechnung 1.6 RC2 which
	allow an unauthorized user to take control of the software.

	Thank's to Brendan Coles for testing and reporting this security issues.


	- Several warnings have been fixed

	- Improved E-Mail function ( mbstring, PEAR::Mail, Mail_Mime is required )

	- Minor bugs have been fixed

	- FPDF was updated to version 1.6


	12.07.2008

	Changes in phpRechnung 1.6 RC2
	---------------------------------------

	There are no new features in this release.

	This is a Bugfix-Only-Release.

	- Addressbook
	  Fix Detail Search non existing fields removed
	  Fix customer info error if message was missing

	- Reports
	  Fix out of memory

	  Note:
	  When creating very large PDF Documents there is a memory limit

	  I have created reports with 457 pages approx. 17130 entries without any problems
	  php.ini -> memory_limit = 16M;

	- Offer / Invoice
	  Show only active positions

	- Update
	  Rename the directory to /updatetable
	  to prevent errors when unpacking the archive on Windows

	- phprechnung.inc.php
	  Easy change of Number Format ( function Format_Number )

	#2001081 - phpRechnung 1.6 RC1 - Update - Offer Status
	  Fix Offer when updating 'OfferStatus'

	#2001200 - phpRechnung 1.6 RC1 - Update - Addressbook - PrintName
	  Fix Addressbook when updating 'PositionName'

	#2004968 -  phpRechnung 1.6 RC1 - Useradministration - admin
	  Fix Usergroup1, Usergroup2 'admin'

	#2010827 - Position - Group
	  Fix wrong table name

	#2012232 - Addressbook Info - Print
	  Fix old / non existing fields

	16.06.2008

	Changes in phpRechnung 1.6 RC1
	---------------------------------------

	- phpRechnung 1.6 RC1 is a total rewrite.

	- phpRechnung is Valid XHTML 1.0 Transitional

	- phpRechnung now has Dutch translation
		Thank's to Jerry van Kranenburg

	- Smarty Template Engine was included ( 2.6.19 )
		Smarty is released under GNU LGPL
		Copyright 2001-2005 New Digital Group, Inc.

	- ADOdb library was included ( adodb498 )
		ADOdb is released under both BSD license and
		Lesser GPL library license.
		(c) 2000-2005 John Lim (jlim#natsoft.com.my).
		All rights reserved.

	- Many improvements

	- Accesskey's
	-------------

		Alt+W - Startpage
		Alt+A - Addressbook
		Alt+P - Position
		Alt+O - Offer
		Alt+I - Invoice
		Alt+N - Creditnote
		Alt+M - Payment
		Alt+C - Cashbook
		Alt+R - Reports
		Alt+S - Configuration
		Alt+U - Super User
		Alt+L - Logout

		Submenu can be reached by pressing Alt+1, for first entry and so on....
		Will not work if you have more than one tab open.

	- phpRechnung Project Home page http://sourceforge.net/projects/phprechnung/

	09.04.2006

	Changes in phpInvoice 1.6 Test3.01
	----------------------------------

	There are no new features in this release; only small corrections.
	$pdfdir is now set to "/tmp/" in phprechnung.inc.php to prevent errors
	when sending invoices or offers by email.
	Format cleaning was done in some files.

	02.10.2005

	Changes in phpInvoice 1.6 Test3
	------------------------------------

	- Now the default language is set to english
		You can change the language in
		/include/sprache.inc.php for the login screen
		and in the SQL or Configuration part for the users.

	- phprechnung.inc.php, $PHPSendMail
		Send email, now you can choose between
		PEAR Mail::Interface -> $PHPSendMail = "1"
		or mail() Function -> $PHPSendMail = "0"
		Default is $PHPSendMail = "0"

	- Fixed PDF-Output, so that long lines
		now can be displayed correctly.
		( Print, E-Mail )
		

offer/edit.php
<?php

/*
	edit.php

	phpRechnung - is easy-to-use Web-based multilingual accounting software.
	Copyright (C) 2001 - 2011 Edy Corak < edy at loenshotel dot de >

	This program is free software; you can redistribute it and/or modify
	it under the terms of the GNU General Public License as published by
	the Free Software Foundation; either version 2 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU General Public License for more details.

	You should have received a copy of the GNU General Public License
	along with this program; if not, write to the Free Software
	Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

require_once("../include/phprechnung.inc.php");
require_once("../include/smarty.inc.php");

CheckUser();
CheckSession();

$ArrayValue = CheckArrayValue($_REQUEST);

foreach($ArrayValue as $key => $val)
{
	$$key = $val;
	$smarty->assign("$key",$val);
}

if(!is_numeric($offerID) || $offerID <= 0 )
{
	die(header("Location: $web"));
}

if(!isset($myID) || !is_numeric($myID) || $myID <= 0 )
{
	$myID = "";
}

if(isset($infoID) && $infoID == '9')
{
	$Searchstring = "&amp;OfferID1=$OfferID1&amp;CustomerID1=$CustomerID1&amp;DateFrom1=$DateFrom1&amp;DateTill1=$DateTill1&amp;Total1=$Total1&amp;Customer1=$Customer1";
	$smarty->assign("Searchstring","$Searchstring");
}

// Assign needed text from language file
//
$smarty->assign("Title","$a[offer] - $a[edit]");
$smarty->assign("Print","$a[print]");
$smarty->assign("Print_Offer","$a[print_offer]");

$smarty->assign("First_Name","$a[firstname]");
$smarty->assign("Last_Name","$a[lastname]");
$smarty->assign("Company_Name","$a[company]");
$smarty->assign("Customer_No","$a[customer_no]");
$smarty->assign("Customer","$a[customer]");
$smarty->assign("Find_Customer","$a[find_customer]");
$smarty->assign("Choose_Customer","$a[choose_customer]");
$smarty->assign("CustMethodOfPayment","$a[cust_method_of_payment]");
$smarty->assign("Date_Till","$a[date_till]");

$smarty->assign("Offer_No","$a[offer_number]");
$smarty->assign("OfferInitials","$a[offer_initials]");
$smarty->assign("CustomerNoInitials","$a[customer_no_initials]");
$smarty->assign("Offer_Amount","$a[offer_amount]");

$smarty->assign("Offer_Tax1","$a[offer_tax1]");
$smarty->assign("Offer_Tax2","$a[offer_tax2]");
$smarty->assign("Offer_Tax3","$a[offer_tax3]");
$smarty->assign("Offer_Subtotal","$a[offer_subtotal]");
$smarty->assign("PositionNew","$a[pos_new]");
$smarty->assign("PositionName","$a[pos_name]");
$smarty->assign("PositionText","$a[pos_text]");
$smarty->assign("PositionQuantity","$a[pos_quantity]");
$smarty->assign("PositionPrice","$a[pos_price]");
$smarty->assign("PositionAmount","$a[pos_amount]");
$smarty->assign("Offer_Note","$a[offer] - $a[note]");
$smarty->assign("Change_Offer","$a[change_offer]");
$smarty->assign("Change","$a[change]");
$smarty->assign("Choose_Message","$a[choose_message]");
$smarty->assign("Choose","$a[choose]");
$smarty->assign("ChangeEntry","$a[entry_changed]");
$smarty->assign("Entry_Canceled","$a[entry_canceled]");

// Database connection
//
DBConnect();

if(!isset($tmpID))
{
	$db->Execute("DELETE FROM {$TBLName}tmp_offer WHERE USERNAME='$_SESSION[Username]'");

	$query = $db->Execute("SELECT OFFERPOSID, MYID, OFFERID, POSITIONID, POS_DESC, POS_QUANTITY, POS_PRICE, POS_GROUP, TAX, TAX_DESC, TAX_MULTI, TAX_DIVIDE FROM {$TBLName}offerpos WHERE OFFERID=$offerID ORDER BY POS_GROUP ASC, OFFERPOSID ASC");

	if (!$query)
		print($db->ErrorMsg());
	else
		foreach($query as $f)
		{
			$PosID = $f['POSITIONID'];
			$Pos_Desc = $f['POS_DESC'];
			$Pos_Quantity = $f['POS_QUANTITY'];
			$Pos_Price = $f['POS_PRICE'];
			$Pos_Group = $f['POS_GROUP'];
			$Tax = $f['TAX'];
			$Tax_Multi = $f['TAX_MULTI'];
			$Tax_Divide = $f['TAX_DIVIDE'];
			$Tax_Desc = $f['TAX_DESC'];

			$query1 = "INSERT INTO {$TBLName}tmp_offer (TMP_OFFERID, MYID, OFFERID, POSITIONID, USERNAME, POS_DESC, POS_QUANTITY, POS_PRICE, POS_GROUP, TAX, TAX_MULTI, TAX_DIVIDE, TAX_DESC)";
			$query1 .= "VALUES (NULL, '$myID', '$offerID', '$PosID', '$_SESSION[Username]', '$Pos_Desc', '$Pos_Quantity', '$Pos_Price', '$Pos_Group', '$Tax', '$Tax_Multi', '$Tax_Divide', '$Tax_Desc')";

			if ($db->Execute($query1) === false)
			{
				die($db->ErrorMsg());
			}
		}
}

$query1 = $db->Execute("SELECT MYID, OFFERID FROM {$TBLName}offer WHERE OFFERID=$offerID");

if (!$query1)
	print($db->ErrorMsg());
else
	foreach($query1 as $f)
	{
		if(empty($myID))
		{
			$myID = $f['MYID'];
		}
		else
		{
			$myID = $myID;
		}
	}

// Get company data from company_settings.inc.php
//
$smarty->assign("Offer_Currency",$CompanyCurrency);
$smarty->assign("Country",$CompanyCountry);
$smarty->assign("TaxFree",$TaxFree);

// Get Offer Information
//
$query2 = $db->Execute("SELECT A.PREFIX, A.TITLE, A.FIRSTNAME, A.LASTNAME, A.ADDRESS, A.COMPANY, A.POSTALCODE, A.PRINT_NAME,
	A.CITY, A.COUNTRY, A.METHODOFPAY, A.MYID, DATE_FORMAT(O.OFFER_DATE,'%d.%m.%Y') AS OFFER_DATE, O.OFFERID, O.TOTAL_AMOUNT, O.MYID,
	O.CREATEDBY, O.STATUS, O.MESSAGEID, O.METHODOFPAYID, O.METHOD_OF_PAY, DATE_FORMAT(O.METHOD_OF_PAY_DATE,'%d.%m.%Y') AS METHOD_OF_PAY_DATE,
	O.TAX1_TOTAL, O.TAX2_TOTAL, O.TAX3_TOTAL, O.TAX4_TOTAL, O.TAX1_DESC, O.TAX2_DESC, O.TAX3_DESC, O.TAX4_DESC, O.SUBTOTAL1, O.SUBTOTAL2, O.SUBTOTAL3, O.SUBTOTAL4, O.NOTE,
	O.CANCELED FROM {$TBLName}addressbook AS A, {$TBLName}offer AS O WHERE A.MYID=$myID AND O.OFFERID=$offerID");

	foreach($query2 as $f)
	{
		if(empty($OfferDate))
		{
			$OfferDate = $f['OFFER_DATE'];
			$smarty->assign("OFFER_DATE",$OfferDate);
		}
		else
		{
			$OfferDate = $OfferDate;
			$smarty->assign("OFFER_DATE",$OfferDate);
		}
		$CreatedBy = $f['CREATEDBY'];
		if(empty($myID))
		{
			$smarty->assign("MYID",$f['MYID']);
		}
		else
		{
			$smarty->assign("MYID",$myID);
		}
		$smarty->assign("TITLE",$f['TITLE']);
		$smarty->assign("PREFIX",$f['PREFIX']);
		$smarty->assign("FIRSTNAME",$f['FIRSTNAME']);
		$smarty->assign("LASTNAME",$f['LASTNAME']);
		$smarty->assign("COMPANY",$f['COMPANY']);
		$smarty->assign("ADDRESS",$f['ADDRESS']);
		$smarty->assign("CITY",$f['CITY']);
		$smarty->assign("POSTALCODE",$f['POSTALCODE']);
		$smarty->assign("COUNTRY",$f['COUNTRY']);
		$smarty->assign("PRINT_NAME",$f['PRINT_NAME']);
		$smarty->assign("CANCELED",$f['CANCELED']);
		$smarty->assign("STATUS",$f['STATUS']);
		if(empty($MethodOfPayment))
		{
			$smarty->assign("NR_METHOD_OF_PAYMENT",$f['METHODOFPAYID']);
		}
		else
		{
			$smarty->assign("NR_METHOD_OF_PAYMENT",$MethodOfPayment);
		}
		if(empty($MethodOfPaymentDate))
		{
			$smarty->assign("METHOD_OF_PAYMENT_DATE",$f['METHOD_OF_PAY_DATE']);
		}
		else
		{
			$smarty->assign("METHOD_OF_PAYMENT_DATE",$MethodOfPaymentDate);
		}
		if(!isset($messageID))
		{
			$smarty->assign("MESSAGEID",$f['MESSAGEID']);
		}
		else
		{
			$smarty->assign("MESSAGEID",$messageID);
		}
		if(empty($Note))
		{
			$smarty->assign("NOTE",$f['NOTE']);
		}
		else
		{
			$smarty->assign("NOTE",$Note);
		}
	}
$PrintD = Print_Date($OfferDate);
$smarty->assign("PrintDate",$PrintD.'-'.$offerID);
$smarty->assign("CurrentOfferID","$offerID");
$smarty->assign("CreatedBy","$CreatedBy");

$posquery = $db->Execute("SELECT P.POSITIONID, P.POS_NAME, T.USERNAME, T.POSITIONID, T.POS_DESC, T.POS_QUANTITY, T.POS_PRICE, T.OFFERID, T.TMP_OFFERID, T.TAX, T.TAX_DIVIDE, T.TAX_MULTI, T.TAX_DESC, T.POS_GROUP FROM {$TBLName}article AS P, {$TBLName}tmp_offer AS T WHERE P.POSITIONID=T.POSITIONID AND T.OFFERID=$offerID ORDER BY T.POS_GROUP ASC, T.POS_DESC ASC");
$numrows = $posquery->RecordCount();

// Calculate positions
//
require_once('../include/pos.inc.php');

// Get the first entry from table 'offer'
//
$query4 = $db->GetRow("SELECT MIN(OFFERID) AS MIN_OFFERID FROM {$TBLName}offer");
if (!$query4)
	print($db->ErrorMsg());
else
	$minOfferID = $query4['MIN_OFFERID'];
	$smarty->assign("MinOfferID","$minOfferID");

// Get the last entry from table 'offer'
//
$query5 = $db->GetRow("SELECT MAX(OFFERID) AS MAX_OFFERID FROM {$TBLName}offer");
if (!$query5)
	print($db->ErrorMsg());
else
	$maxOfferID = $query5['MAX_OFFERID'];

	$smarty->assign("MaxOfferID","$maxOfferID");

// If we are not on first page then display
// first page, previous page link
//
if ($offerID > $minOfferID)
{
	$CurrentOfferID = $offerID - 1;
	$smarty->assign('PrevOfferID', "$CurrentOfferID");
}

// If we are not on the last page then display
// next page, last page link
//
if ($offerID < $maxOfferID)
{
	$CurrentOfferID = $offerID + 1;
	$smarty->assign('NextOfferID', "$CurrentOfferID");
}

// Get the method of payment from database
//
$query6 = $db->Execute("SELECT METHODOFPAYID, DESCRIPTION FROM {$TBLName}methodofpay ORDER BY DESCRIPTION ASC");

// Get the message from database
//
$query7 = $db->Execute("SELECT MESSAGEID, DESCRIPTION FROM {$TBLName}message ORDER BY DESCRIPTION ASC");

// If an error has occurred, display the error message
//
if (!$query6 && !$query7)
	print $db->ErrorMsg();
else
	foreach($query6 as $result6)
	{
		$PaymentData[] = $result6;
	}

	if(isset($PaymentData))
		$smarty->assign("PaymentData",$PaymentData);

	foreach($query7 as $result7)
	{
		$MessageData[] = $result7;
	}

	if(isset($MessageData))
		$smarty->assign("MessageData",$MessageData);

	if(isset($_SESSION['Username']) && $_SESSION['Username'] != $root && $_SESSION['Usergroup1'] != $admingroup_1 && $_SESSION['Usergroup2'] != $admingroup_2 && $_SESSION['Username'] != $CreatedBy)
	{
		$_SESSION['LastSite'] = $_SERVER['PHP_SELF'].'?'.$_SERVER['QUERY_STRING'];
		$_SESSION['logoutid'] = "5";
		Header("Location: $web/login/sustart.php?$sessname=$sessid");
	}
	else
	{
		// Save last page visited by user
		//
		UserSite();
		$smarty->display('offer/edit.tpl');
	}

?>
